src/SwedbankPay.Sdk/SwedbankPayHttpClient.cs

```diff
@@ -0,0 +1,170 @@
+        ///     Send a HttpPost and Process HttpResponse for a url
+        /// </summary>
+        /// <typeparam name="TResponse"></typeparam>
+        /// <param name="url"></param>
+        /// <param name="payload"></param>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="InvalidOperationException"></exception>
+        /// <exception cref="HttpRequestException"></exception>
+        /// <exception cref="HttpResponseException"></exception>
+        internal async Task<TResponse> HttpPost
+            <TResponse>(Uri url, object payload)
+            where TResponse : new()
+        {
+            var httpRequestMessage = CreateHttpRequestMessage(HttpMethod.Post, url, payload);
+            return await SendHttpRequestAndProcessHttpResponse<TResponse>(httpRequestMessage);
+        }
+
+
+        /// <summary>
+        ///     Send the HttpRequest and Process HttpResponse
+        /// </summary>
+        /// <typeparam name="TResponse"></typeparam>
+        /// <param name="httpMethod"></param>
+        /// <param name="url"></param>
+        /// <param name="payload"></param>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="InvalidOperationException"></exception>
+        /// <exception cref="HttpRequestException"></exception>
+        /// <exception cref="HttpResponseException"></exception>
+        /// <returns></returns>
+        internal async Task<TResponse> SendHttpRequestAndProcessHttpResponse
+            <TResponse>(HttpMethod httpMethod,
+                        Uri url,
+                        object payload = null)
+            where TResponse : new()
+        {
+            var httpRequestMessage = CreateHttpRequestMessage(httpMethod, url, payload);
+
+            return await SendHttpRequestAndProcessHttpResponse<TResponse>(httpRequestMessage);
+        }
+
+
+        /// <summary>
+        ///     Send the HttpRequest and Process HttpResponse
+        /// </summary>
+        /// <typeparam name="TResponse"></typeparam>
+        /// <param name="httpRequest"></param>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="InvalidOperationException"></exception>
+        /// <exception cref="HttpRequestException"></exception>
+        /// <exception cref="HttpResponseException"></exception>
+        /// <returns></returns>
+        internal async Task<TResponse> SendHttpRequestAndProcessHttpResponse
+            <TResponse>(HttpRequestMessage httpRequest)
+            where TResponse : new()
+        {
+            var httpResponse = await this.client.SendAsync(httpRequest);
+
+            string BuildErrorMessage(string httpResponseBody)
+            {
+                return
+                    $"{httpRequest.Method}: {httpRequest.RequestUri} failed with error code {httpResponse.StatusCode} using bearer token {httpRequest.Headers.Authorization.Parameter}. Response body: {httpResponseBody}";
```

Comment:
`httpRequest.Headers.Authorization.Parameter` will throw a null pointer exception since we're not checking that it exists.

Which begs the question: Should we check in the `SwedbankPayHttpClient`'s constructor if an authorization token exists? Should we set it in the constructor ourselves? Should we still set it in the `HttpClient` which is passed into the constructor like we're currently (trying) to do now?

My opinion is that we should keep this information as close to the `SwedbankPayHttpClient` somehow so that it's easier to use without having to remember all these details. I'm not a fan of modifying the `HttpClient` after the fact because it tends to lead to awkward code and/or race conditions as well, so there's that.

Let's discuss.